core/arch/arm/kernel/thread.c

```diff
@@ -958,13 +1063,17 @@ void thread_init_threads(void)
 	}
 }
 
-void thread_clr_thread_core_local(void)
+void __nostackcheck thread_clr_thread_core_local(void)
 {
 	size_t n = 0;
+	struct thread_core_local *tcl = thread_core_local;
 
 	for (n = 0; n < CFG_TEE_CORE_NB_CORE; n++)
-		thread_core_local[n].curr_thread = -1;
-	thread_core_local[0].flags |= THREAD_CLF_TMP;
+		tcl[n].curr_thread = -1;
+
+	tcl[0].flags |= THREAD_CLF_TMP;
```

Comment:
We could do this for all `thread_core_local`. That way we wouldn't need to update `flags` from `reset_secondary()` in `~/work/op-tee/optee_os2/core/arch/arm/kernel/entry_a32.S`.

The call in `init_secondary_helper()` (which looks a bit fishy) could be removed too.

Reply:
Agreed.